Makefile

```diff
@@ -105,40 +113,78 @@ check-scripts:
 	@luac -p $(shell find lua/* -iname '*.lua')
 
 check:
+ifdef MONO
+	@echo
+	@echo "Compiling in debug mode with Mono..."
+	@$(MSBUILD) -t:Build -restore -p:Configuration="Debug - Mono" -p:TargetPlatform=$(TARGETPLATFORM) -p:TargetFramework=netstandard2.1 -p:Mono=true -p:DefineConstants="MONO"
+	@echo
+	@echo "Checking runtime assemblies..."
```

Comment:
These are mostly the same except for mono vs dotnet - how about we define a single `$(RUNTIME_LAUNCHER)` in the preamble that takes `mono --debug` or `$(DOTNET)` to reduce duplication?

Reply:
WHITELISTED_CORE_ASSEMBLIES can then be set to the mono vs core specific list in the same if/else that assigns RUNTIME_LAUNCHER.